.github/workflows/workflow.yaml

```diff
@@ -1,76 +1,42 @@
-name: CI / CD
+name: Workflow
 
-on: [push, pull_request]
+on: 
+  push:
+    branches:
+      - 'main'
+      - 'ci-*'
+  pull_request:
 
 jobs:
 
-  # bump-tag:
-  #   if: github.ref == 'refs/heads/main'
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #       with:
-  #         fetch-depth: '0'
-  #     - id: bump_version
-  #       name: Bump version and push tag
-  #       uses: anothrNick/github-tag-action@1.46.0
-  #       env:
-  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  #         WITH_V: true
-  #   outputs:
-  #     new_tag: ${{ steps.bump_version.outputs.new_tag }}
-
-  build-linux:
-    runs-on: ubuntu-latest
+  ci:
+    name: CI
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        platform: [gcc]
+        os: [ubuntu-latest, macos-latest]
     steps:
       - uses: actions/checkout@v3
       - name: Setup Ninja
         uses: ashutoshvarma/setup-ninja@master
         with:
           version: 1.10.2
       - name: Build
-        env:
-          CC: ${{ steps.vars.outputs.cc }}
-          CXX: ${{ steps.vars.outputs.cxx }}
-          MATRIX_PLATFORM: ${{ matrix.platform }}
-        run: .github/scripts/build-linux.sh
+        id: build
+        run: .github/scripts/build-${{ runner.os }}.sh
```

Comment:
Can we collapse the windows job into this job by adding the build script into the matrix?
e.g.:
```yml
matrix:
    include:
      - os: 'ubuntu-latest'
        script: 'build-linux.sh'
      - os: 'macos-latest'
        script: 'build-macos.sh'
      - os: 'windows-latest'
        script: 'build-msvc.ps1'
```

Reply:
I considered that, but for windows we currently build with an additional platform matrix, so I wasn't sure how we should deal with that. Probably the easiest way would be to drop support for 32 bit windows builds, but its quite nice being able to build for the original architecture

Reply:
Let's leave 32-bit builds enabled, if not for checking our pointer arithmetic is correct.

You mean the `ilammy/msvc-dev-cmd@v1.4.1` step? That can be made conditional on the os using
```yml
      - uses: ilammy/msvc-dev-cmd@v1.4.1
        if: contains(matrix.os, 'windows')
        with:
          arch: ${{ matrix.platform }}
```

Reply:
And the platforms can then only be added to the Windows items.

```yml
    include:
      - os: 'ubuntu-latest'
        script: '.github/scripts/build-linux.sh'
      - os: 'macos-latest'
        script: '.github/scripts/build-macos.sh'
      - os: 'windows-latest'
        script: '.github\\scripts\\build-msvc.ps1'
        platform: 'x86'
      - os: 'windows-latest'
        script: '.github\\scripts\\build-msvc.ps1'
        platform: 'amd64'
```

Reply:
no, I mean for windows we build x86 and x64. Ah, just saw your other comment, yep that should owrk